chain/processor.go

```diff
@@ -41,7 +41,7 @@ func (b *StatelessBlock) Execute(
 		f       = fetcher.New(im, numTxs, b.vm.GetStateFetchConcurrency())
 		e       = executor.New(numTxs, b.vm.GetTransactionExecutionCores(), MaxKeyDependencies, b.vm.GetExecutorVerifyRecorder())
 		ts      = tstate.New(numTxs * 2) // TODO: tune this heuristic
-		results = make([]*Result, numTxs)
+		results = make([]*Result, 0)
```

Comment:
Do we want a 2d array with a length of `numTxs` instead? or flatten all the results into one array? not sure of the consequences of doing the flatten approach right now